sparta/sparta/ports/SyncPort.hpp

```diff
@@ -255,6 +262,11 @@ namespace sparta
             sparta_assert(clk_->isPosedge(), "Posedge check failed in port:" << getLocation());
 
             Clock::Cycle send_cycle = clk_->currentCycle() + send_delay_cycles;
+            
+            if(allow_slide) {
+                send_cycle += sync_in_port_->getNumInFlight();
+            }
```

Comment:
Wonder if this code is best placed in the inport's `send_` method or better yet `computeSendToReceiveTickDelay_`.  Thoughts?

Reply:
Ok, let me know how you want to proceed with this.

Reply:
I had a look at `computeNextAvailableCycleForSend` which should compute the next available clock cycle for a packet to be sent. However, I am observing a rather curious behavior, when it is used:

```
21: top.arch.memory_cpu0.crossbar.ports.out_arbiter0: SEND @22 allow_slide=1 # 0x5565f39b6070, 
21: top.arch.memory_cpu0.ports.in_crossbar_to_llc: RECEIVE SCHEDULED @22(0)  # 0x5565f39b6070

21: top.arch.memory_cpu0.crossbar.ports.out_arbiter0: SEND @24 allow_slide=1 # 0x5565f39b6070,
21: top.arch.memory_cpu0.ports.in_crossbar_to_llc: RECEIVE SCHEDULED @24(1)  # 0x5565f39b6070

21: top.arch.memory_cpu0.crossbar.ports.out_arbiter0: SEND @25 allow_slide=1 # 0x5565f39b6070, prevDataSendCycle: 24, sendCycle: 25
21: top.arch.memory_cpu0.ports.in_crossbar_to_llc: RECEIVE SCHEDULED @25(2)  # 0x5565f39b6070
```

As you can see, the first sending at 21 is to be received at clock cycle 22. With `allow_slide` set, the second sending at 21 through the same port should be scheduled in 23. However, `computeNextAvailableCycleForSend` reports 24. Any succeeding sending through the same port is the previous one +1 which is correct again. The reason for the 24 is the call to `Clock::calculateClockCrossingDelay` which returns an incorrect delay for the send in the middle.

I have only a limited understanding, what happens in Sparta. Since I am leaving SyncPort now and trying to adapt `Clock`, I fear that I am going to break more than I fix. In addition, I am venturing into the area of clock domain crossing, which is again unknown terrain for me.

Since this PR disregards any clock domain crossing by just counting the packets in flight and adjusting the send delay accordingly, I would not recommend merging it. Consider, for instance, that the receiver could have a clock twice as fast as the sender and hence could accept 2 packets per clock cycle from the perspective of the sender. This PR does not take this into the account.

For me personally, counting the number of packets in flight works because I just have 1 clock domain. I would keep the PR to remember that the SyncPort has some issues with sliding packets. If allow_slide is removed, it renders this PR obsolete. Alternatively, someone who is comfortable with the inner workings of the SyncPort, should have a look into it.

Reply:
I have removed the access to the number of packets in flight and use `computeNextAvailableCycleForSend` with a few changes.
Again, Terms and Conditions apply: I am by no means an expert in the field. I see the advantage of using the method `computeNextAvailableCycleForSend` as it computes the next available cycle considering clock domain crossings. However, I do not have test cases to prove the correct behavior.